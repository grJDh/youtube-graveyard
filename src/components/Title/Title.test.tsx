import { render, screen } from "@testing-library/react";
import Title from "./Title";
import { describe, expect, it } from "vitest";
import { BrowserRouter } from "react-router-dom";

describe("#App", () => {
  it("renders App", async () => {
    render(<Title />, { wrapper: BrowserRouter });
    const headerElement = screen.getByRole("banner");
    const titleElement = screen.getByText(/YouTube Graveyard/i);
    const logoElement = screen.getByAltText(/logo/i);
    const linkElement = screen.getByRole("link");

    expect(headerElement).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
    expect(logoElement).toBeInTheDocument();
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "/");
    expect(linkElement).toHaveAttribute("aria-label", "Return to the Start");

    expect(headerElement).toMatchSnapshot();
  });
});
