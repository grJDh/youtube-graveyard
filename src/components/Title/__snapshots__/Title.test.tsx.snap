// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`#App > renders App 1`] = `
<header
  class="title"
>
  <a
    aria-label="Return to the Start"
    href="/"
  >
    <img
      alt="Logo"
      src="/youtube-graveyard/src/assets/grave_main.svg"
    />
    <h1>
      YouTube Graveyard
    </h1>
  </a>
</header>
`;
